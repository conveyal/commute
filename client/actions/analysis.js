import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
