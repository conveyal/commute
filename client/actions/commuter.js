import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
