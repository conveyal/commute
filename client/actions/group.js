import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
