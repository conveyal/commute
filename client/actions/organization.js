import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
