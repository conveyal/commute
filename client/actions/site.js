import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
