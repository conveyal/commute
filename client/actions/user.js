import {createAction} from 'redux-actions'

// login / logout
export const login = createAction('log in')
export const logout = createAction('log out')
