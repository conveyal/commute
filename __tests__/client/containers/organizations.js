/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Button} from 'react-bootstrap'
import {Provider} from 'react-redux'

import {expectDeleteOrganization} from '../../test-utils/actions'
import {makeMockStore, mockStores} from '../../test-utils/mock-data'

import Organizations from '../../../client/containers/organizations'

describe('Container > Organizations', () => {
  it('Organizations View loads', () => {
    // mount component
    const tree = mount(
      <Provider store={makeMockStore(mockStores.init)}>
        <Organizations />
      </Provider>
    )
    expect(mountToJson(tree.find(Organizations))).toMatchSnapshot()
  })

  it('Organizations View loads with one organization', () => {
    // mount component
    const tree = mount(
      <Provider store={makeMockStore(mockStores.oneSimpleOrganization)}>
        <Organizations />
      </Provider>
    )
    expect(mountToJson(tree.find(Organizations))).toMatchSnapshot()
  })

  it('Delete Organization', () => {
    const mockStore = makeMockStore(mockStores.oneSimpleOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the organizations view
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Organizations />
      </Provider>
    )

    // When the user clicks the delete button for an existing organization
    // And the user confirms the Confirm Deletion dialog
    const deleteButton = tree.find('table').find(Button).last()
    deleteButton.simulate('click')

    expectDeleteOrganization(mockStore.getActions())
  })
})
