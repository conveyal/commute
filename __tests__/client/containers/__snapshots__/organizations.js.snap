exports[`Container > Organizations Delete Organization 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/agency/agency-3",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > Organizations Organizations View loads 1`] = `
"<div class=\"container\">
  <div class=\"row\">
    <div class=\"col-xs-12\">
      <h2>
        <i class=\"fa fa-flag fa-fw \"></i><span>Mock Agency</span> </h2>
      <div class=\"btn-group\">
        <button to=\"/agency/agency-1/edit\" type=\"button\" class=\"btn btn-warning\">Edit</button>
        <button type=\"button\" class=\"btn btn-danger\">Delete</button>
      </div>
      <h3>

        <!-- react-text: 11 -->Organizations<!-- /react-text -->
        <button to=\"/agency/agency-1/organization/create\" type=\"button\" class=\"pull-right btn btn-default\"><span>Create a new organization</span> 
          <i class=\"fa fa-shield fa-fw \"></i>
        </button>
      </h3>
      <p>

        <!-- react-text: 16 -->An organization is a collection of sites<!-- /react-text -->
        <i class=\"fa fa-building fa-fw \"></i>

        <!-- react-text: 18 -->and commuters<!-- /react-text -->
        <i class=\"fa fa-users fa-fw \"></i>

        <!-- react-text: 20 -->.<!-- /react-text -->
      </p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
                        <col class=\"\">
                          <col class=\"\">
              </colgroup>
              <thead>
                <tr>
                  <th class=\"\" data-is-only-head=\"false\" style=\"text-align: left; display: none;\" data-field=\"_id\">
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Name\" style=\"text-align: left;\" data-field=\"name\">

                    <!-- react-text: 37 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Sites\" style=\"text-align: left;\" data-field=\"sites\">

                    <!-- react-text: 40 -->Sites<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Groups\" style=\"text-align: left;\" data-field=\"groups\">

                    <!-- react-text: 43 -->Groups<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Analyses\" style=\"text-align: left;\" data-field=\"analyses\">

                    <!-- react-text: 46 -->Analyses<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 49 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
                        <col class=\"\">
                          <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td data-toggle=\"collapse\" colspan=\"6\" class=\"react-bs-table-no-data\">There is no data to display</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Container > Organizations Organizations View loads with one organization 1`] = `
"<div class=\"container\">
  <div class=\"row\">
    <div class=\"col-xs-12\">
      <h2>
        <i class=\"fa fa-flag fa-fw \"></i><span>An agency</span> </h2>
      <div class=\"btn-group\">
        <button to=\"/agency/agency-3/edit\" type=\"button\" class=\"btn btn-warning\">Edit</button>
        <button type=\"button\" class=\"btn btn-danger\">Delete</button>
      </div>
      <h3>

        <!-- react-text: 11 -->Organizations<!-- /react-text -->
        <button to=\"/agency/agency-3/organization/create\" type=\"button\" class=\"pull-right btn btn-default\"><span>Create a new organization</span> 
          <i class=\"fa fa-shield fa-fw \"></i>
        </button>
      </h3>
      <p>

        <!-- react-text: 16 -->An organization is a collection of sites<!-- /react-text -->
        <i class=\"fa fa-building fa-fw \"></i>

        <!-- react-text: 18 -->and commuters<!-- /react-text -->
        <i class=\"fa fa-users fa-fw \"></i>

        <!-- react-text: 20 -->.<!-- /react-text -->
      </p>
      <div class=\"react-bs-table-container\">
        <div class=\"react-bs-table\" style=\"height: 100%;\">
          <div class=\"react-bs-container-header table-header-wrapper\">
            <table class=\"table table-hover table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none; min-width: NaNpx;\">
                  <col class=\"\" style=\"min-width: NaNpx;\">
                    <col class=\"\" style=\"min-width: NaNpx;\">
                      <col class=\"\" style=\"min-width: NaNpx;\">
                        <col class=\"\" style=\"min-width: NaNpx;\">
                          <col class=\"\" style=\"min-width: NaNpx;\">
              </colgroup>
              <thead>
                <tr>
                  <th class=\"\" data-is-only-head=\"false\" style=\"text-align: left; display: none;\" data-field=\"_id\">
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Name\" style=\"text-align: left;\" data-field=\"name\">

                    <!-- react-text: 37 -->Name<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Sites\" style=\"text-align: left;\" data-field=\"sites\">

                    <!-- react-text: 40 -->Sites<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Groups\" style=\"text-align: left;\" data-field=\"groups\">

                    <!-- react-text: 43 -->Groups<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Analyses\" style=\"text-align: left;\" data-field=\"analyses\">

                    <!-- react-text: 46 -->Analyses<!-- /react-text -->
                    <div></div>
                  </th>
                  <th class=\"\" data-is-only-head=\"false\" title=\"Tools\" style=\"text-align: left;\" data-field=\"undefined\">

                    <!-- react-text: 49 -->Tools<!-- /react-text -->
                    <div></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class=\"react-bs-container-body\" style=\"height: 100%;\">
            <table class=\"table table-bordered\">
              <colgroup>
                <col class=\"\" style=\"display: none;\">
                  <col class=\"\">
                    <col class=\"\">
                      <col class=\"\">
                        <col class=\"\">
                          <col class=\"\">
              </colgroup>
              <tbody>
                <tr class=\"\">
                  <td class=\"\" style=\"text-align: left; display: none;\">organization-1</td>
                  <td class=\"\" style=\"text-align: left;\">
                    <a>Mock Organization</a> 
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>0</div>
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>0</div>
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div>0</div>
                  </td>
                  <td class=\"\" style=\"text-align: left;\">
                    <div class=\"btn-group\">
                      <button to=\"/organization/organization-1/edit\" type=\"button\" class=\"btn btn-warning\">Edit</button>
                      <button type=\"button\" class=\"btn btn-danger\">Delete</button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
